app/questionnaire/navigation.py

```diff
@@ -34,22 +35,32 @@ def build_navigation(self, current_group_id, current_group_instance):
 
         navigation = []
 
+        last_group_id = SchemaHelper.get_last_group_id(self.survey_json)
+
         for group in filter(lambda x: 'hide_in_navigation' not in x, self.survey_json['groups']):
             first_location = Location(group['id'], 0, group['blocks'][0]['id'])
 
             logger.debug("building frontend navigation", group_id=group['id'])
 
             skip_group = self._should_skip_group(current_group_instance, group)
+            last_block_in_current_group = group['blocks'][-1]
+            is_summary_or_confirmation = last_block_in_current_group and \
+                last_block_in_current_group['type'] in ('Summary', 'Confirmation')
 
             if not skip_group:
                 repeating_rule = SchemaHelper.get_repeat_rule(group)
                 if repeating_rule:
                     logger.debug("building repeating navigation", group_id=group['id'])
                     navigation.extend(self._build_repeating_navigation(repeating_rule, group, current_group_id,
                                                                        current_group_instance))
-                else:
+                elif last_group_id == group['id'] and is_summary_or_confirmation and self.routing_path:
+                    logger.debug("building navigation", group_id=group['id'])
+                    if len(self.completed_blocks) > 0 and set(self.routing_path[:-1]).issubset(self.completed_blocks):
```

Comment:
For an empty routing path `self.routing_path[:-1]` == `[]`. So we'd be checking if `completed_blocks` is a subset of `[]` where we have more than zero completed blocks. So this should be safe.